Sources/ZcashLightClientKit/Rust/ZcashRustBackend.swift

```diff
@@ -281,46 +281,46 @@ struct ZcashRustBackend: ZcashRustBackendWelding {
     }
 
     @DBActor
-    func getTransparentBalance(account: Int32) async throws -> Int64 {
-        guard account >= 0 else {
-            throw ZcashError.rustGetTransparentBalanceNegativeAccount(Int(account))
+    func getTransparentBalance(account: Zip32Account) async throws -> Int64 {
+        guard account.index >= 0 else {
+            throw ZcashError.rustGetTransparentBalanceNegativeAccount(Int(account.index))
         }
 
```

Comment:
It's no longer necessary to check that the index is in range, because that is ensured by the `Zip32AccountIndex` constructor. Also, delete all occurrences of `rustGetTransparentBalanceNegativeAccount` since it can no longer happen.

```suggestion
    func getTransparentBalance(accountIndex: Zip32AccountIndex) async throws -> Int64 {
```

Reply:
Updated